File: src/lowlevel/server/icefsoperators/icefsDestroy.go
/*
 * @Author: Tan90degrees tangentninetydegrees@gmail.com
 * @Date: 2023-03-11 07:18:32
 * @LastEditors: Tan90degrees tangentninetydegrees@gmail.com
 * @LastEditTime: 2023-04-17 15:51:58
 * @FilePath: /icefs/src/lowlevel/server/icefsoperators/icefsDestroy.go
 * @Description:
 *
 * Copyright (C) 2023 Tan90degrees <tangentninetydegrees@gmail.com>.
 */
package icefsoperators

import (
	"context"
	"icefs-server/icefserror"
	pb "icefs-server/icefsgrpc"
	"icefs-server/icefsthrift"
)

func (s *IcefsServer) doIcefsDestroy(hostName string, reqInfo string) (status int32, resInfo string) {
	status = icefserror.ICEFS_EOK
	resInfo = "Bye"
	return
}

func (s *IcefsGRpcServer) DoIcefsDestroy(ctx context.Context, req *pb.IcefsDestroyReq) (*pb.IcefsDestroyRes, error) {
	var res pb.IcefsDestroyRes

	res.Status, res.Info = s.server.doIcefsDestroy(req.HostName, req.Info)

	return &res, nil
}

func (s *IcefsThriftServer) DoIcefsDestroy(ctx context.Context, req *icefsthrift.IcefsDestroyReq) (*icefsthrift.IcefsDestroyRes, error) {
	var res icefsthrift.IcefsDestroyRes

	res.Status, res.Info = s.server.doIcefsDestroy(req.HostName, req.Info)

	return &res, nil
}

File: src/lowlevel/client/icefsoperators/icefsDestroy.cpp
/*
 * @Author: Tan90degrees tangentninetydegrees@gmail.com
 * @Date: 2023-03-30 04:19:29
 * @LastEditors: Tan90degrees tangentninetydegrees@gmail.com
 * @LastEditTime: 2023-04-21 02:40:41
 * @FilePath: /icefs/src/lowlevel/client/icefsoperators/icefsDestroy.cpp
 * @Description:
 *
 * Copyright (C) 2023 Tan90degrees <tangentninetydegrees@gmail.com>.
 */
#include <stdio.h>

#include <string>

#include "icefsClient.hpp"
#include "icefsClientUtils.hpp"

void IcefsClient::DoIcefsDestroy(void *userData) {
  ICEFS_PR_FUNCTION;

  switch (this->clientConfig.linkType) {
    case ICEFS_LINK_USE_GRPC: {
      icefsgrpc::IcefsDestroyReq req;
      icefsgrpc::IcefsDestroyRes res;
      grpc::ClientContext ctx;

      req.set_host_name("Hello");
      req.set_info("I'm the client.");

      grpc::Status status = gRpcClient->DoIcefsDestroy(&ctx, req, &res);
      if (status.ok() && !res.status()) {
        std::cout << res.info() << std::endl;
      } else {
        ICEFS_PR_ERR_STATUS;
      }
      break;
    }

    case ICEFS_LINK_USE_THRIFT: {
      icefsthrift::IcefsDestroyReq req;
      icefsthrift::IcefsDestroyRes res;

      req.__set_host_name("Hello");
      req.__set_info("I'm the client.");

      icefsThriftConn *thriftConn = thriftClientPool->GetIcefsThriftConn();
      thriftConn->thriftClient->DoIcefsDestroy(res, req);
      thriftClientPool->PutIcefsThriftConn(thriftConn);
      if (!res.status) {
        std::cout << res.info << std::endl;
      } else {
        ICEFS_PR_ERR_STATUS;
      }
      break;
    }

    default:
      break;
  }
}
